kyuubi-metrics/src/main/scala/org/apache/kyuubi/metrics/PrometheusReporterService.scala

```diff
@@ -100,4 +112,46 @@ class PrometheusReporterService(registry: MetricRegistry)
       }
     }
   }
+
+  private def createPrometheusServletWithLabels(labels: Map[String, String]): HttpServlet = {
+    new HttpServlet {
+      override def doGet(request: HttpServletRequest, response: HttpServletResponse): Unit = {
+        try {
+          response.setContentType("text/plain;charset=utf-8")
+          response.setStatus(HttpServletResponse.SC_OK)
+          response.getWriter.print(getMetricsSnapshot(labels))
+        } catch {
+          case e: IllegalArgumentException =>
+            response.sendError(HttpServletResponse.SC_BAD_REQUEST, e.getMessage)
+          case e: Exception =>
+            warn(s"GET ${request.getRequestURI} failed: $e", e)
+            throw e
+        }
+      }
+
+      // ensure TRACE is not supported
+      override protected def doTrace(req: HttpServletRequest, res: HttpServletResponse): Unit = {
+        res.sendError(HttpServletResponse.SC_METHOD_NOT_ALLOWED)
+      }
+    }
+  }
+
+  private def getMetricsSnapshot(labels: Map[String, String]): String = {
+    val metricsSnapshotWriter = new java.io.StringWriter
+    val contentType = TextFormat.chooseContentType(null)
+    TextFormat.writeFormat(contentType, metricsSnapshotWriter, bridgeRegistry.metricFamilySamples())
+    val labelStr = labelString(labels)
+    metricsSnapshotWriter.toString.split("\n").map { line =>
+      if (line.startsWith("#")) {
+        line
+      } else {
+        val Array(metrics, value) = line.split("\\s+", 2)
+        s"""$metrics${labelStr} $value"""
```

Comment:
a little bit hacky, but given the feature is disabled by default, let's merge it in first.

Reply:
changed to

```
       line.split("\\s+", 2) match {
          case Array(metrics, rest) => s"""$metrics${labelStr} $rest"""
          case _ => line
        }
```